src/Kunstmaan/FormBundle/Tests/unit/Entity/FormSubmissionFieldTypes/FileFormSubmissionFieldTest.php

```diff
@@ -79,6 +80,55 @@ public function testGetSubmissionTemplate()
     }
 
     public function testUpload()
+    {
+        $object = $this->object;
+
+        $file = $this->getMockBuilder(UploadedFile::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $file->expects($this->any())
+            ->method('getPathname')
+            ->willReturn(__FILE__);
+
+        $object->file = $file;
+
+        $form = $this->getMockBuilder(Form::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $builder = $this->getMockBuilder(FormBuilder::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $request = new Request();
+
+        $container = $this->getMockBuilder(Container::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $container->expects($this->any())
+            ->method('getParameter')
+            ->willReturn('whatever');
+
+        $filesystem = $this->getMockBuilder(Filesystem::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $container->expects($this->any())
+            ->method('has')
+            ->willReturn(true);
+
+        $container->expects($this->any())
+            ->method('get')
+            ->willReturn($filesystem);
+
+        $object->onValidPost($form, $builder, $request, $container);
+        $this->assertNull($object->upload('...', '..', $filesystem));
+        $object->upload(__DIR__ . '/../../Resources/assets/', __DIR__ . '/../../Resources/assets/', $filesystem);
+    }
+
+    public function testUploadBC()
```

Comment:
Also add a `@expectedDeprecation` assertion to make sure the required deprecation is triggered

Reply:
Done